js/ui/control/attribution.js

```diff
@@ -49,6 +49,15 @@ Attribution.prototype = util.inherit(Control, {
             }
         }
 
+        // remove any entries that are duplicates or strict substrings of another entry.
+        attributions.sort();
```

Comment:
@lucaswoj i figured sorting the list first means that, in the filter function for the `i-th` item, you can safely start the loop at `i+1`.  Without this, you have to check every item, and I think it gets a little weird to correctly filter out exact duplicates.